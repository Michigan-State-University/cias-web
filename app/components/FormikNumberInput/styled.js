import styled from 'styled-components';

import { colors } from 'theme';

export const ErrorText = styled.div`
  color: ${colors.flamingo};
  font-size: 0.8rem;
  font-weight: bold;
`;
