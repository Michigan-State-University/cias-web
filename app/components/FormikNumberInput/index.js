/**
 *
 * FormikNumberInput
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { useField } from 'formik';

import { Input } from 'components/Input';
import Text from 'components/Text';
import { formatIncompletePhoneNumber } from 'libphonenumber-js/mobile';

import Column from 'components/Column';
import { ErrorText } from './styled';

function FormikNumberInput({
  formikKey,
  placeholder,
  type,
  label,
  inputProps,
  countryCode,
  ...columnStyleProps
}) {
  const [field, meta, helpers] = useField(formikKey);
  const { value, onBlur } = field;
  const { error, touched } = meta;
  const hasError = touched && error;
  const { setValue, setTouched } = helpers;

  const handleChange = inputValue => {
    setValue(
      formatIncompletePhoneNumber(inputValue.target.value, countryCode ?? 'US'),
    );
    setTouched(true, false);
  };

  return (
    <Column {...columnStyleProps}>
      <Text mb={5} width="fit-content">
        {label}
      </Text>
      <Input
        mb={hasError ? 5 : null}
        placeholder={placeholder}
        value={value}
        name={formikKey}
        onChange={handleChange}
        onBlur={onBlur}
        hasError={hasError}
        keyboard={type}
        {...inputProps}
      />
      {hasError && <ErrorText>{error.toString()}</ErrorText>}
    </Column>
  );
}

FormikNumberInput.propTypes = {
  formikKey: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  countryCode: PropTypes.string,
  type: PropTypes.string,
  label: PropTypes.string,
  inputProps: PropTypes.object,
};

export default memo(FormikNumberInput);
