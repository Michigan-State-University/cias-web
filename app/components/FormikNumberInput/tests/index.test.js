/**
 *
 * Tests for FormikInput
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React, { Fragment } from 'react';
import { render } from '@testing-library/react';
import { Formik } from 'formik';

import FormikNumberInput from '../index';

describe('<FormikNumberInput />', () => {
  let props;
  beforeAll(() => {
    props = {
      formikKey: 'test',
      placeholder: 'test',
      type: 'text',
      label: 'Label',
    };
  });
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Formik initialValues={{ test: '' }} onSubmit={jest.fn()}>
        {() => (
          <Fragment>
            <FormikNumberInput {...props} />
          </Fragment>
        )}
      </Formik>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const { container } = render(
      <Formik initialValues={{ test: '' }} onSubmit={jest.fn()}>
        {() => (
          <Fragment>
            <FormikNumberInput {...props} />
          </Fragment>
        )}
      </Formik>,
    );
    expect(container).toMatchSnapshot();
  });
});
