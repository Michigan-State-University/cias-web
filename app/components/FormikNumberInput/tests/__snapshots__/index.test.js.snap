// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormikNumberInput /> Should render and match the snapshot 1`] = `
<div>
  <div
    class="Column-sc-1n6divy-0 SJAYV"
  >
    <p
      class="Text-sc-2lza2m-0 drfDFr"
      width="fit-content"
    >
      Label
    </p>
    <input
      class="Input-sc-19txxuq-0 lPrqp"
      name="test"
      placeholder="test"
      type="text"
      value=""
    />
  </div>
</div>
`;
